docs/user-guide/app_dependencies.md

```diff
@@ -0,0 +1,127 @@
+# Application dependencies
+
+Starting from Argo CD v2.9, you can natively model dependencies between Applications. This new mechanism is intended to replace the prior way of modeling dependencies when using the app-of-apps pattern with sync waves on the Application custom resources.
```

Comment:
I'm not a huge fan of burning version numbers like this (2.9) into commit contents. They make predictions that often fail.

I'd _somewhat_ favor "if you're reading this, and using the version related to this documentation" or something like that. Or even a placeholder macro and letting the doc build system fill it in, and then at release branch point burn the versions into the main branch documentation.

Reply:
```suggestion
Argo CD lets you natively model dependencies between Applications. This mechanism is intended to replace the prior way of modeling dependencies when using the app-of-apps pattern with sync waves on the Application custom resources.
```

Reply:
I agree. Does above suggestion sound better?